packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1044 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+        {{/each}}`
+    });
+
+    this.registerComponent('child-non-block', {
+      template: 'Child: {{item}}.'
+    });
+
+    let items = emberA(['Tom', 'Dick', 'Harry']);
+
+    this.render('{{non-block items=items}}', { items });
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.context.get('items').pushObject('Sergio'));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.get('items').shiftObject());
+
+    this.assertText('In layout. [Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.set('items', emberA(['Tom', 'Dick', 'Harry'])));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+  }
+
+  ['@skip specifying classNames results in correct class'](assert) {
+    let clickyThing;
+
+    this.registerComponent('some-clicky-thing', {
+      ComponentClass: Component.extend({
+        tagName: 'button',
+        classNames: ['foo', 'bar'],
+        init() {
+          this._super(...arguments);
+          clickyThing = this;
+        }
+      })
+    });
+
+    this.render(strip`
+      {{#some-clicky-thing classNames="baz"}}
+        Click Me
+      {{/some-clicky-thing}}`
+    );
+
+    let expectedClassNames = ['ember-view', 'foo', 'bar', 'baz'];
+
+    assert.ok(this.$('button').is('.foo.bar.baz.ember-view'), `the element has the correct classes: ${this.$('button').attr('class')}`);
+    assert.deepEqual(clickyThing.get('classNames'), expectedClassNames, 'classNames are properly combined');
+    assert.deepEqual(this.$('button').attr('class').split(' '), expectedClassNames, 'all classes are set 1:1 in DOM');
+
+    this.runTask(() => this.rerender());
+
+    assert.ok(this.$('button').is('.foo.bar.baz.ember-view'), `the element has the correct classes: ${this.$('button').attr('class')} (rerender)`);
+    assert.deepEqual(clickyThing.get('classNames'), expectedClassNames, 'classNames are properly combined (rerender)');
+    assert.deepEqual(this.$('button').attr('class').split(' '), expectedClassNames, 'all classes are set 1:1 in DOM (rerender)');
+  }
+
+  ['@skip specifying custom concatenatedProperties avoids clobbering'](assert) {
+    let clickyThing;
+    this.registerComponent('some-clicky-thing', {
+      ComponentClass: Component.extend({
+        concatenatedProperties: ['blahzz'],
+        blahzz: ['blark', 'pory'],
+        init() {
+          this._super(...arguments);
+          clickyThing = this;
+        }
+      })
+    });
+
+    this.render(strip`
+      {{#some-clicky-thing blahzz="baz"}}
+        Click Me
+      {{/some-clicky-thing}}`
+    );
+
+    assert.deepEqual(clickyThing.get('blahzz'), ['blark', 'pory', 'baz'], 'property is properly combined');
```

Comment:
we can just render the values in the template instead of using the JS instance. (`{{#each blahzz ...}}`)